rules/windows/powershell/powershell_script/posh_ps_reg_query_registry.yml

```diff
@@ -0,0 +1,38 @@
+title: Potential Configuration And Service Reconnaissance Via Powershell
+id: 064060aa-09fb-4636-817f-020a32aa7e9e
+related:
+    - id: 970007b7-ce32-49d0-a4a4-fbef016950bd
+      type: similar
+status: experimental
+description: Detects the usage of Powershell in order to query reconnaissance information from the registry. Adversaries may interact with the Windows registry to gather information about credentials, the system, configuration, and installed software.
+references:
+    - https://github.com/redcanaryco/atomic-red-team/blob/f339e7da7d05f6057fdfcdd3742bfcf365fee2a9/atomics/T1012/T1012.md
+author: frack113
+date: 2023/07/02
+tags:
+    - attack.discovery
+    - attack.t1012
+    - attack.t1007
+logsource:
+    product: windows
+    category: ps_script
+    definition: 'Requirements: Script Block Logging must be enabled'
+detection:
+    selection_cmdlet:
+        ScriptBlockText|contains:
+            - 'Get-Item'
+            - 'Get-ChildItem'
+    selection_flag:
+        ScriptBlockText|contains: '-Path'
+    selection_key:
+        ScriptBlockText|contains:
+            - 'CurrentVersion\Windows'
+            - 'winlogon\'
+            - 'CurrentVersion\ShellServiceObjectDelayLoad'
+            - 'CurrentVersion\Run' # Also covers the strings "RunOnce", "RunOnceEx" and "runServicesOnce"
+            - 'CurrentVersion\Policies\Explorer\Run'
+            - 'currentcontrolset\services'
+    condition: all of selection_*
+falsepositives:
+    - Unknown
```

Comment:
I'd expect quite some scripts to contain the combination of strings over the whole ScriptBlockText. Maybe a regex could be used to increase the quality of the rule by setting a proximity of get-item to the path.

Reply:
I prefer to avoid regex.
And since it's powershell script, it's complicated to find a pattern with this very permissive language.
If we get many FP the rule can be set to low.

Reply:
This rule won't work in PowerShell script as it is. As @phantinuss pointed out `Get-ChildItem` and `-Path` are virtually used in every admin script. So we assume that they're always there which only leaves the paths. Which can be used in different context. For example to set the key or to simply get its value for further changes.

Since the rule is stating recon it needs to look for successive calls to those registry. Or simply combine it like `Get-ChildItem -Path XXX`

Either way this is a `low` at best and should be transferred to "hunting"

Reply:
Move to hunting is one way. But writing a regex for this rule still would work in my opinion. It wouldn't find all occurrences as @frack113 said with PowerShell having too many ways to do stuff, but you could go for a best-effort rule. As is, the rule also misses some shorthands like "gci" for Get-ChildItem. There are many more, I would need to look them up.

Reply:
So 6 `(Get-Item|gci|Get-ChildItem).*-Path.*(\\key_to_find\\)` ?
Don't want a All In One hell regex .

Reply:
the `.*` is the bad part and also not restricting. So I'd suggest something looking like:
```
(Get-Item|gci|Get-ChildItem).{1,64}-Path.{1,64}\\(path1|path2|path3)\\
```
The 64 is just a first guess. The shorter we choose it, the less FP we get. But it should be long enough to match the recon scripts.

Reply:
We can create both rules. One for hunting and one for regular matching. The regex should be applied case-insensitive because of the paths and different possible `gEt-cHiLDiTeM` casings